plugins/modules/digital_ocean_droplet_info.py

```diff
@@ -0,0 +1,245 @@
+        name: "New York 1"
+        sizes:
+          - s-1vcpu-1gb
+          - s-1vcpu-1gb-intel
+          - s-1vcpu-2gb
+          - s-1vcpu-2gb-intel
+          - s-2vcpu-2gb
+          - s-2vcpu-2gb-intel
+          - s-2vcpu-4gb
+          - s-2vcpu-4gb-intel
+          - s-4vcpu-8gb
+          - c-2
+          - c2-2vcpu-4gb
+          - s-4vcpu-8gb-intel
+          - g-2vcpu-8gb
+          - gd-2vcpu-8gb
+          - s-8vcpu-16gb
+          - m-2vcpu-16gb
+          - c-4
+          - c2-4vcpu-8gb
+          - s-8vcpu-16gb-intel
+          - m3-2vcpu-16gb
+          - g-4vcpu-16gb
+          - so-2vcpu-16gb
+          - m6-2vcpu-16gb
+          - gd-4vcpu-16gb
+          - so1_5-2vcpu-16gb
+          - m-4vcpu-32gb
+          - c-8
+          - c2-8vcpu-16gb
+          - m3-4vcpu-32gb
+          - g-8vcpu-32gb
+          - so-4vcpu-32gb
+          - m6-4vcpu-32gb
+          - gd-8vcpu-32gb
+          - so1_5-4vcpu-32gb
+          - m-8vcpu-64gb
+          - c-16
+          - c2-16vcpu-32gb
+          - m3-8vcpu-64gb
+          - g-16vcpu-64gb
+          - so-8vcpu-64gb
+          - m6-8vcpu-64gb
+          - gd-16vcpu-64gb
+          - so1_5-8vcpu-64gb
+          - m-16vcpu-128gb
+          - c-32
+          - c2-32vcpu-64gb
+          - m3-16vcpu-128gb
+          - m-24vcpu-192gb
+          - g-32vcpu-128gb
+          - so-16vcpu-128gb
+          - m6-16vcpu-128gb
+          - gd-32vcpu-128gb
+          - m3-24vcpu-192gb
+          - g-40vcpu-160gb
+          - so1_5-16vcpu-128gb
+          - m-32vcpu-256gb
+          - gd-40vcpu-160gb
+          - so-24vcpu-192gb
+          - m6-24vcpu-192gb
+          - m3-32vcpu-256gb
+          - so1_5-24vcpu-192gb
+          - so-32vcpu-256gb
+          - m6-32vcpu-256gb
+          - so1_5-32vcpu-256gb
+        slug: nyc1
+      size:
+        available: true
+        description: Basic
+        disk: 25
+        memory: 1024
+        price_hourly: 0.00744
+        price_monthly: 5.0
+        regions:
+          - ams2
+          - ams3
+          - blr1
+          - fra1
+          - lon1
+          - nyc1
+          - nyc2
+          - nyc3
+          - sfo1
+          - sfo3
+          - sgp1
+          - tor1
+        slug: s-1vcpu-1gb
+        transfer: 1.0
+        vcpus: 1
+      size_slug: s-1vcpu-1gb
+      snapshot_ids: []
+      status: active
+      tags:
+        - tag1
+      vcpus: 1
+      volume_ids: []
+      vpc_uuid: 123-abc-567a
+  type: list
+'''
+
+from traceback import format_exc
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.digitalocean.plugins.module_utils.digital_ocean import DigitalOceanHelper
+from ansible.module_utils._text import to_native
+
+
+def run(module):
+    rest = DigitalOceanHelper(module)
+
+    if module.params["id"]:
+        path = "droplets/" + module.params["id"]
+        return_codes = (200, 404)
+    else:
+        path = "droplets"
+        return_codes = (200)
+
+    response = rest.get(path)
```

Comment:
In my original review, I forgot that the droplet API is paginated. This line of code will fail to retrieve complete data if the user has more than 20 droplets (at least I think the default page size is 20).

So we actually need to use  https://github.com/ansible-collections/community.digitalocean/blob/f93b52c7cd0ed78b7864a7f0d650143e43d59e3e/plugins/module_utils/digital_ocean.py#L98 here when the id is not set.

Reply:
Fixed.